File: AngularWebApiSample/src/AngularWebApiSample/ClientApp/src/app/api/models/CombinedQueryModel.ts


import { SimpleModel } from './SimpleModel';

export interface ICombinedQueryModel {
    $type?: string;
    id?: number;
    simpleModels?: SimpleModel[];
    
}

export class CombinedQueryModel implements ICombinedQueryModel {
    public $type: string;
    public id: number;
    public simpleModels: SimpleModel[];
    
    constructor(initObj?: ICombinedQueryModel) {
        this.$type = 'AngularWebApiSample.Models.CombinedQueryModel, AngularWebApiSample';
        if(initObj) {
            this.id = initObj.id || 0;
            this.simpleModels = initObj.simpleModels || [];
            
        }
        else {
            this.id = 0;
            this.simpleModels = [];
            
        }
    }
}

File: src/AngularWebApiSample/Models/CombinedQueryModel.cs
using System.Collections.Generic;
using AngularWebApiSample.Attributes;

namespace AngularWebApiSample.Models;

[GenerateFrontendType]
public class CombinedQueryModel
{
    public int Id { get; set; }

#pragma warning disable MA0016 // Prefer return collection abstraction instead of implementation
    public List<SimpleModel> SimpleModels { get; internal set; } = new List<SimpleModel>();
#pragma warning restore MA0016 // Prefer return collection abstraction instead of implementation

    public string FirstName { get; set; } = string.Empty;

    public string? MiddleName { get; set; }

    public string LastName { get; set; } = string.Empty;
}
